### DISCUSSION TITLE
wire:navigate remove query parameters when prefetch page with mouse pressed (or hover)

### DISCUSSION THREAD

**Discussion by serge-kilimoff** [livewire/livewire#7715] [7. Bugs (archived)] [answered]
### Livewire version

3.3.5

### Laravel version

10.40.0

### Which PHP version are you using?

PHP 8.3

### Steps To Reproduce

`<a href="http://localhost/page?arg1=1&arg2=2" wire:navigate>Link</a>`

On press click (before release), livewire execute query `http://localhost/page` and on release, execute query `http://localhost/page?arg1=1&arg2=2`. Same thing with hover prefetch.

### Repository or Wirebox URL

https://wirebox.app/b/43ke1

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by mgmbh** [comment]
I can confirm this.

Two small changes are needed to fix this:

In the prefetchHtml() function, change `let path = destination.pathname` to `let path = destination.pathname + destination.search` 

In the storeThePrefetchedHtmlForWhenALinkIsClicked() function, change `let state = prefetches[destination.pathname]` to `let state = prefetches[destination.pathname + destination.search]`

Be aware that it is not recommended to edit this file on your machine, and that your changes will be overwritten when a new update is available. But for me, this is an easy fix until the guys had the time to look at this bug.

**Comment by aftabkhaliq86** [comment]
I am facing the same issue .
and if  wire:navigate.hover is in loop its only triggers for the fist time only 
` @forelse ($chatLists as $chat_list)
                                    <div class=" {{$conversationId == $chat_list?->user?->id?'bg-light':'' }}">
                                        <a wire:navigate.hover href="{{route('agent.chat',['id'=> $chatModuleId ,'conversation' => $chat_list?->user?->id,'orderby'=>$sort,'agentId'=>$agentId != 0 ?$agentId:''])}}" class="text-body cursor-pointer bg-light @if (!$loop->last) border-bottom @endif h-24"
                                        </div>
                                        @endforelse`

**Comment by joshhanley** [comment, marked as answer]
We're in the process of investigating all wire:navigate issues and I believe this has been fixed in PR #8143 so I'm going to close this. If you're still experiencing problems, please open a new discussion. Thanks!
